WalletCore/Classes/Core/Service/OffchainService.swift

```diff
@@ -36,9 +36,58 @@ public class OffchainService {
     public func trade(amount: Decimal, asset: LWAssetModel, forAsset: LWAssetModel) -> Observable<ApiResult<LWModelOffchainResult>> {
         let channelAsset = amount > 0 ? forAsset : asset
         
-        let pair = dependency.authManager.assetPairs
-            .requestAssetPair(baseAsset: asset, quotingAsset: forAsset)
+        return createOffchainOperationObervable(forChannelAsset: channelAsset) { [dependency] decryptedKeyObservable in
+            let pair = dependency.authManager.assetPairs
+                .requestAssetPair(baseAsset: asset, quotingAsset: forAsset)
+
+            let operationObservable = Observable
+                .zip(decryptedKeyObservable, pair.filterSuccess().filterNil())
+                .map{decryptedKey, pair in LWPacketOffchainTrade.Body(
+                    asset: channelAsset.identity,
+                    assetPair: pair.identity,
+                    prevTempPrivateKey: decryptedKey,
+                    volume: amount
+                    )}
+                .flatMapLatest{ [dependency] body in
+                    return dependency.authManager.offchainTrade.request(withData: body)
+                }
+                .shareReplay(1)
+            let errorsObservable: Observable<[AnyHashable : Any]> = pair.filterSuccess().filter{ $0 == nil }.map{ _ in ["Message": "There is no asset pair for your request."]}
+            return (operationObservable, errorsObservable)
+        }
+    }
+    
+    public func finalizePendingRequests(refresh: Observable<Void>) -> Disposable {
+        return refresh
+            .processPendingRequests(dependency)
+            .subscribe()
+    }
+    
+    public func cashOutSwift(amount: Decimal, fromAsset asset: LWAssetModel, toBank bankName: String, iban: String, bic: String, accountHolder: String, accountHolderAddress: String) -> Observable<ApiResult<LWModelOffchainResult>> {
         
+        return createOffchainOperationObervable(forChannelAsset: asset) { [dependency] decryptedKeyObservable in
+            let operationObservable = decryptedKeyObservable
+                .map { decryptedKey in LWPacketOffchainCashOutSwift.Body(
+                    amount: amount,
+                    asset: asset.identity,
+                    bankName: bankName,
+                    iban: iban,
+                    bic: bic,
+                    accountHolder: accountHolder,
+                    accountHolderAddress: accountHolderAddress,
+                    prevTempPrivateKey: decryptedKey)}
+                .flatMapLatest { [dependency] data in
+                    return dependency.authManager.offchainCashOutSwift.request(withData: data)
+                }
+                .shareReplay(1)
+            let errorsObservable: Observable<[AnyHashable : Any]> = Observable.never()
```

Comment:
Please handle API errors accordingly, otherwise in case of error the app won't show them and the loader won't disappear